worker/uniter/runner/context/context.go

```diff
@@ -984,6 +984,12 @@ func (ctx *HookContext) Prepare() error {
 	return nil
 }
 
+// Commit flushes the context.
+// Implements jujuc.HookContext.unitCacheContext, part of runner.Context.
+func (ctx *HookContext) Commit() error {
+	return ctx.Flush(ctx.hookName, nil)
```

Comment:
I have a question here. 

I you read the implementation for Flush, there is a bit of code in there (a defer to be exact) that takes care of (pending) reboot requests. I am not sure if we should process such requests when the charm asks for a commit as the hook execution hasn't technically completed yet. Thoughts?

Of course, changing the Flush signature so we can indicate whether it is a commit or final flush would break most of the tests so if that's the way to go, I would recommend waiting for #11204 to land first (adds extra flush tests) so we don't have to deal with conflicts.

Reply:
We do need to skip the Reboot, I'll wait for #11204 to land.